mujoco_warp/_src/types.py

```diff
@@ -1315,6 +1315,7 @@ class Data:
     ximat: Cartesian orientation of body inertia                (nworld, nbody, 3, 3)
     xanchor: Cartesian position of joint anchor                 (nworld, njnt, 3)
     xaxis: Cartesian joint axis                                 (nworld, njnt, 3)
+    geom_sameframe: Indicate whether to compute geom pose       (ngeom,)
```

Comment:
I don't think the semantics are the same as C MuJoCo so I'm not sure we should call it `geom_sameframe` - it's just that the geom body is the worldbody - it might be in a different frame than the world body, right?  And we would still want to apply this optimization.

Also we want the user to understand that we will not write to `geom_xpos` in some scenarios.  This might be surprising to them, e.g. imagine if they store temp data in `geom_xpos` and then expect it to be fixed after running `step`.

So my suggestion would be call this `geom_skip` and then clearly document its behavior, e.g.:

```py
geom_skip: skip calculating `geom_xpos` and `geom_xmat` during step, reuse previous value
```

WDYT?